_pytest/junitxml.py

```diff
@@ -265,6 +265,8 @@ def __init__(self, logfile, prefix):
         ], 0)
         self.node_reporters = {}  # nodeid -> _NodeReporter
         self.node_reporters_ordered = []
+        self.global_node_reporter = None
```

Comment:
I'm not sure you need this... my impression is that keeping the properties in `global_properties` is enough and simplifies the code quite a bit.